docs/DeveloperGuide.md

```diff
@@ -1,33 +1,211 @@
+components.
+
+
+### Architecture
+
+Given below is a quick overview of main components and how they interact with each other.
+
+**Main components of the architecture**
+
+**`AthletiCLI`** is in charge of the app launch and shut down.
+
+The bulk of the AthletiCLI’s work is done by the following components, with each of them corresponds to a package:
+
+* [`UI`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/ui): The UI of AthletiCLI.
+* [`Storage`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/storage): Reads data from, and writes data to, the hard disk.
+* [`Data`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/data): Holds the data of AthletiCLI in memory.
+* [`Commands`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/commands): The command executors.
+
+[`Exceptions`](https://github.com/AY2324S1-CS2113-T17-1/tp/tree/master/src/main/java/athleticli/exceptions) represents exceptions used by multiple other components.
+
+### UI Component
+
+### Storage Component
+
+### Data Component
+
+### Commands Component
+
+## Implementation
+
+### Diet Management in AthletiCLI
+
+#### [Implemented] Setting Up, Editing, Deleting, Listing, and Finding Diets
+
+Regardless of the operation you are performing on diets (setting up, editing, deleting, listing, or finding), the process follows a general five-step pattern in AthletiCLI:
+
+1. **Input Processing**: The user's input is passed through AthletiCLI to the Parser Class. Examples of user inputs include:
+    - `add-diet calories/500 protein/20 carb/50 fat/10 datetime/2021-09-01 06:00` for adding a diet.
+    - `edit-diet 1 calories/500 protein/20 carb/50 fat/10 datetime/2021-09-01 06:00` for editing a diet.
+    - `delete-diet 1` for deleting a diet.
+    - `list-diet` for listing all diets.
+    - `find-diet 2021-09-01` for finding diets of a particular date.
+
+2. **Command Identification**: The Parser Class identifies the type of diet operation and passes the necessary parameters.
+
+3. **Command Creation**: An instance of the corresponding command class is created (e.g., AddDietCommand, EditDietCommand, etc.) and returned to AthletiCLI.
+
+4. **Command Execution**: AthletiCLI executes the command, interacting with the data instance of DietList to perform the required operation.
+
+5. **Result Display**: A message is returned post-execution and passed through AthletiCLI to the UI for display to the user.
+
+By following these general steps, AthletiCLI ensures a streamlined process for managing diet-related tasks.
+
+### [Implemented] Setting Up of Diet Goals
+
+This following sequence diagram show how the 'set-diet-goal' command works:
+
+<p  align="center" width="100%">
+  <img width="80%" src="images/setDietGoalUmlSequenceDiagram.svg" alt="'set-diet-goal' Sequence Diagram"/>
```

Comment:
Maybe it would help to include variable names in the diagram? I.e. it may not be immediately clear which object's execute() is called unless we look into the code/make educated guesses